query-service-impl/src/test/java/org/hypertrace/core/query/service/postgres/QueryRequestToPostgresSQLConverterTest.java

```diff
@@ -0,0 +1,1212 @@
+    Filter endTimeFilter = createTimeFilter("Span.end_time_millis", Operator.LT, 1570744906673L);
+    QueryRequest queryRequest =
+        QueryRequest.newBuilder()
+            .addAggregation(
+                createAliasedFunctionExpression(
+                    "DISTINCTCOUNT", "Span.id", "distinctcount_span_id"))
+            .setFilter(
+                Filter.newBuilder()
+                    .setOperator(Operator.AND)
+                    .addChildFilter(startTimeFilter)
+                    .addChildFilter(endTimeFilter)
+                    .build())
+            .setLimit(15)
+            .build();
+
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        queryRequest,
+        "select count(distinct encode(span_id, 'hex')) FROM public.\"span-event-view\""
+            + " where "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "and ( start_time_millis > 1570658506605 and end_time_millis < 1570744906673 )"
+            + " limit 15",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithDistinctCountAggregationAndGroupBy() {
+    Filter startTimeFilter =
+        createTimeFilter("Span.start_time_millis", Operator.GT, 1570658506605L);
+    Filter endTimeFilter = createTimeFilter("Span.end_time_millis", Operator.LT, 1570744906673L);
+    QueryRequest queryRequest =
+        QueryRequest.newBuilder()
+            .addSelection(createColumnExpression("Span.id"))
+            .addGroupBy(createColumnExpression("Span.id"))
+            .addAggregation(
+                createAliasedFunctionExpression(
+                    "DISTINCTCOUNT", "Span.id", "distinctcount_span_id"))
+            .setFilter(
+                Filter.newBuilder()
+                    .setOperator(Operator.AND)
+                    .addChildFilter(startTimeFilter)
+                    .addChildFilter(endTimeFilter)
+                    .build())
+            .addOrderBy(
+                createOrderByExpression(
+                    createAliasedFunctionExpression(
+                        "DISTINCTCOUNT", "Span.id", "distinctcount_span_id"),
+                    SortOrder.ASC))
+            .setLimit(15)
+            .build();
+
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        queryRequest,
+        "select encode(span_id, 'hex'), count(distinct encode(span_id, 'hex')) FROM public.\"span-event-view\""
+            + " where "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "and ( start_time_millis > 1570658506605 and end_time_millis < 1570744906673 )"
+            + " group by span_id order by count(distinct span_id) limit 15",
```

Comment:
Note: we should use alias in order by. Like in this example, we might have two eval for  `count(distinct span_id)` in order clause and `count(distinct encode(span_id, 'hex'))` in select clause as expressions are different. However, we can confirm this by explain_plan.